### PR TITLE
[DRFT-252] Check for drift after system checkin

### PR DESCRIPTION
**Pull request by CodeHeeler**
After archiving a new hsp, check for baselines associated with
this system, check for any drift from those baselines, and, if found,
emit signal to Notifications app via kafka to trigger alert.

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

**Comment by CodeHeeler** [comment]
This one is much rougher, still currently WIP, just getting it out so we can work together on it tomorrow more easily.  I have a number of questions for things I'm less familiar with:
--how do we actually emit a message to kafka
--where should the two new service interfaces live, kerlescan or within hsp (started within hsp for ease of testing since all lacking here is the constants for the endpoints)
--how do I get the auth key?  I see it's in the headers in views, but how do I get that here, or is it needed or...?
--what endpoint name are we associating with this new baselines field to handle this query on the association?

File: historical_system_profiles/archiver.py [hunk 1]
```diff
@@ -81,6 +83,36 @@ def _archive_profile(data, ptc, logger):
             "wrote %s to historical database (inv id: %s, captured_on: %s)"
             % (hsp.id, hsp.inventory_id, hsp.captured_on)
         )
+    # After the new hsp is saved, we need to check for any reason to alert Notifications
+    _check_need_for_notifications(host["id"])
```
Comment (petrblaho) [inline historical_system_profiles/archiver.py:87]: I am not really sure we need to do that for both branches of if-else (lines 69 and 77) as with duplicate one we don't write anything into db only with new one we do.

File: historical_system_profiles/archiver.py [hunk 2]
```diff
@@ -81,6 +83,36 @@ def _archive_profile(data, ptc, logger):
...
+
+
+def _check_need_for_notifications(inventory_id)
```
Comment (petrblaho) [inline historical_system_profiles/archiver.py:90]: I am not sure about naming - we are not only checking for the need to send the notification we are also emitting the message if the need is there...

What about `_check_and_send_notifications` or something similar?

**Comment by petrblaho** [comment]
> This one is much rougher, still currently WIP, just getting it out so we can work together on it tomorrow more easily. I have a number of questions for things I'm less familiar with:
> --how do we actually emit a message to kafka
> --where should the two new service interfaces live, kerlescan or within hsp (started within hsp for ease of testing since all lacking here is the constants for the endpoints)
> --how do I get the auth key? I see it's in the headers in views, but how do I get that here, or is it needed or...?

To be honest I don't know if it is needed - we are doing that request on behalf of the app itself not for the user who is logged it.... 

Anyway, for request object from inside of Flask app you can always do `from flask import request`

> --what endpoint name are we associating with this new baselines field to handle this query on the association?

I don't have answers to other questions yet.

**Comment by CodeHeeler** [comment]
Accompanying kerlescan constants which will need to be tagged and then have this pr update dependency to it first:
https://github.com/RedHatInsights/kerlescan/pull/19

**Comment by CodeHeeler** [comment]
Pulling this portion out of draft into "real" PR.  The `_emit_notifications_signal` method is stubbed out here but will be completed in a separate PR that addresses only Kafka stuff, adding the Producer, etc.
The `service_auth_key` is a placeholder pending how we decide to handle the shared secret for calling from our app without an actual relevant user.

**Comment by petrblaho** [comment]
I created https://github.com/RedHatInsights/historical-system-profiles-backend/pull/127 with some code updates (as I can't update this PR directly)

**Comment by CodeHeeler** [comment]
Closing; @petrblaho created another WIP PR building on this as #127  while I was away, and I then have been adding to that in #131

**Pull request closed without merging**